Src/Support/Google.Apis.Auth/OAuth2/AwsExternalAccountCredential.AwsSignedSubjectToken.cs

```diff
@@ -145,6 +145,7 @@ string GetHashedCanonicalRequest()
                         canonicalRequest.Append(headerPair.Value);
                         canonicalRequest.Append(newLine);
                     }
+                    canonicalRequest.Append(newLine);
```

Comment:
OK, I got what was wrong with my tests:

- I was previously adding the line breaks before appending the next element instead of after appending the previous one, so...
- The code in line 152 looked like `canonicalRequest.Append(newLine).Append(signedHeaders);` which  added the missing line break.
- But the code in line 134 looked like `canonicalRequest.Append(newLine).Append(verificationRequestUrl.GetQueryStringForAwsCanonicalRequest()).Append(newLine);` which is pretty ugly (appending line breaks before and after on account of the for loop adding the headers right after).

So I changed the code to add the line breaks after each element, not realizing that I was removing this line break with that. And I managed to cherry pick that change out of the last (and possibly more) local nuget package that I created, which I did rougly two hours before merging the original PR. I'm sorry about this! But I'm glad I understand what happened.

The unit test that I have added will catch any such changes. And I've also run my manual tests with the code on this PR, and it's good.